x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/common/H3CartesianUtil.java

```diff
@@ -51,24 +44,24 @@ public final class H3CartesianUtil {
             }
         }
         for (int res = 2; res <= H3.MAX_H3_RES; res++) {
-            CACHED_H3.put(NORTH[res], getCoordinates(NORTH[res]));
-            CACHED_H3.put(SOUTH[res], getCoordinates(SOUTH[res]));
+            CACHED_H3.put(H3.northPolarH3(res), getCoordinates(H3.northPolarH3(res)));
+            CACHED_H3.put(H3.southPolarH3(res), getCoordinates(H3.southPolarH3(res)));
         }
     }
 
     private static final double[] NORTH_BOUND = new double[16];
```

Comment:
Perhaps the size should be based on `H3.MAX_H3_RES + 1` instead of `16`

Reply:
done